### ISSUE TITLE
Code of conduct etc

### ISSUE THREAD

**Issue by lsjroberts** [Rotonde/beaker#49]
While this is a pretty nascent project, it seems sensible to me to put in place things like a code of conduct from the outset.

_Contributor note: all these can be done here https://github.com/Rotonde/beaker/community_

## Code of conduct

Github recommends the [Contributor Covenant](https://www.contributor-covenant.org/version/1/4/code-of-conduct.html). I have no qualms with it if no-one else does?

- https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/

## Contribution guidelines

This one is probably less important, but we could put anything regarding code design decisions etc here.

- https://help.github.com/articles/setting-guidelines-for-repository-contributors/
- https://github.com/opengovernment/opengovernment/blob/master/CONTRIBUTING.md
- https://github.com/atom/atom/blob/master/CONTRIBUTING.md

## License

This is actually pretty important. The default license of code on github technically only permits users to view the code and not modify it (if I'm not mistaken).

I usually go with MIT, but there are [other options worth considering](https://choosealicense.com/).

> The MIT License is a permissive license that is short and to the point. It lets people do anything they want with your code as long as they provide attribution back to you and don’t hold you liable.

> The GNU GPLv3 is a copyleft license that requires anyone who distributes your code or a derivative work to make the source available under the same terms, and also provides an express grant of patent rights from contributors to users.

## Issue and pull request templates

These help with triaging and ensuring we get enough information to understand the issue and decide how to proceed.

- https://help.github.com/articles/creating-an-issue-template-for-your-repository/
- https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/

---

Thoughts?

**Comment by cblgh** [comment]
added in #50

**Comment by lsjroberts** [comment]
There's more than just the code of conduct here :)

**Comment by cblgh** [comment]
they call me speedy gonzales you know, my bad!

**Comment by neauoire** [comment]
This is outside my area of expertise a little bit, but I usually go with CC. Wasn't there a lot of drama with MIT involving React? I failed to understand what that was about but Rotonde should pretty much be as close as possible to GNU, like do what thou will. I'm open to anything really, as long as it lets people feel confident that nobody will go after them, that we always make sure that 100% of the code is from Rotonde contributors and not external libraries.

**Comment by webdesserts** [comment]
@neauoire the drama with react was that they used a custom license with a patent clause in it. They eventually settled on MIT after WordPress and others said they would stop using React due to the patent clause. So it wasn't really an issue with MIT, MIT was the solution to the issue. That said I know nothing about licenses 🙃

**Comment by neauoire** [comment]
Thanks for the clarification :)
Let's go with MIT!

**Comment by lsjroberts** [comment]
MIT is good. Creative Commons recommends against using their licenses for software as that is not what they were intended for (last I checked).

The React issue is interesting as now there is technically less protection for anyone using React. But it's all a bit opaque and untested.

**Comment by frozenpandaman** [comment]
Agreed, MIT is nice and permissive. I also like GPL, very similar but disallows projects leveraging/using the code in commercial or proprietary ways, etc. So it sort of forces the open-ness on all derivative works. :) Depends on your philosophy/aim though, of course.

We'll want to add the code of conduct & license to the new client and user repos as well at some point.
